Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,9 +58,17 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    { // check for exact format "hh:mm:ss.zzz, voltage, current"
+        QRegExp rxItem("(^([0-1]\\d|2[0-3]):[0-5]\\d:[0-5]\\d\\.\\d\\d\\d, \\d+\\.\\d+, (-\\d|\\d)+\\.\\d+$)");
```

Comment:
Fancy, I would split this up into a time regex, voltage regex, current regex, and put them together, easier to understand. Especially since the team has alot of newbies, regexp isn't a common beginner concept, something like that should be friendly to read for working in the team in the future..

```
hourRegexp = "blahblahblah";
analogRegexp = "blahblahblah";
batDataRx = hourregexp + analogRegexp + analogRegexp;
```

BTW you're the first to take a regexp approach :)